cluster-state-service/canary/pkg/tests/util/validate.go

```diff
@@ -0,0 +1,39 @@
+package util
+
+import (
+	"time"
+
+	"github.com/blox/blox/cluster-state-service/canary/pkg/wrappers"
+	"github.com/pkg/errors"
+)
+
+func ValidateECSContainerInstanceAndGetInstanceARN(ecsWrapper wrappers.ECSWrapper, clusterName string) (string, error) {
+	// Takes some time for EC2 instance to come up and for the ECS agent to
+	// register the container instance with ECS.
+	// Retry ECS list call once every minute for 10 minutes
+	var instanceARN string
+	found := false
+	for i := 0; i < 10; i++ {
+		time.Sleep(1 * time.Minute)
+		instances, err := ecsWrapper.ListContainerInstances(&clusterName)
+		if err != nil {
+			return "", err
+		}
+		if len(instances) > 1 {
```

Comment:
Currently only one instance is launched for each test. Will be changing it (was getting something extremely basic in place). And everything was a single file before I moved it around into functions. So due to the copy-pasting, i havent generalized it. Will do :)